lib/src/boilerplate_suggestors/boilerplate_utilities.dart

```diff
@@ -119,3 +120,14 @@ void migrateClassToMixin(ClassDeclaration node, YieldPatch yieldPatch,
   propsAndStateClassNamesConvertedToNewBoilerplate[originalPublicClassName] =
       newMixinName;
 }
+
+/// Returns the name of the props class for a given [factoryDeclaration].
+String getPropsClassNameFromFactoryDeclaration(
+    TopLevelVariableDeclaration factoryDeclaration) {
+  return factoryDeclaration.variables.type.childEntities
+      .whereType<TypeArgumentList>()
+      .single
+      .arguments
+      .single
+      .toSource();
```

Comment:
#nit This could break if the type argument isn't specified. Also, avoid single
```dart
  return factoryDeclaration.variables.type.childEntities
      .whereType<TypeArgumentList>()
      .firstOrNull
      ?.arguments
      ?.first
      ?.toSource();
```

```dart
extension<E> on Iterable<E> {
  E firstOrNull => isEmpty ? null : first;
}